doc/extensions/acrbuildext/azext_acrbuildext/build_task.py

```diff
@@ -0,0 +1,79 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import LongRunningOperation
+from ._utils import (
+    arm_deploy_template_build_task_create,
+    validate_managed_registry
+)
+from .build import _check_image_name
+
+BUILD_TASKS_NOT_SUPPORTED = 'Build Tasks are only supported for managed registries.'
+
+
+def acr_build_task_create(
+    cmd,
+    client,
+    build_task_name,
+    registry_name,
+    context,            
+    source_branch,  
+    image_name,
+    git_access_token,
+    resource_group_name=None,
+    docker_file_path='Dockerfile',
+    os_type="Linux",
+    cpu=1):
+    
+    registry, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, BUILD_TASKS_NOT_SUPPORTED)
+
+    LongRunningOperation(cmd.cli_ctx)(
+            arm_deploy_template_build_task_create(
+                cmd.cli_ctx,
+                build_task_name,
+                registry_name,
+                registry.location,
+                resource_group_name,
+                context,
+                source_branch,              
+                _check_image_name(image_name),
+                docker_file_path,
+                git_access_token,
+                os_type,
+                cpu)
+        )
+    return client.get(resource_group_name, registry_name, build_task_name)
```

Comment:
It looks strange to me that we are creating a build task with a build step here but we only show the build task to users after creation. After discussions I understood the need here but a command called `build-task create` shouldn't create another resource implicitly and people had complained about this in the past.

I assume we will have another set of commands such as

```
acr build-task step create
acr build-task step show
acr build-task step delete
acr build-task step list
```

for build step specifically. So

1. From my understanding, build step will also have different types, which should be properly supported in `acr build-task step create` command.
2. If build step has to be created with a build task, I would recommend that `acr build-task create` by default only create the build task and provide additional flags to create build steps of different types.

/cc @northtyphoon @sajayantony @mnltejaswini

Reply:
Now we want to "hide" build step from CLI. We are exploring the option to use yaml config so things may change.